tuning/trackers/aimstack_tracker.py

```diff
@@ -20,6 +23,49 @@
 from .tracker import Tracker
 from tuning.config.tracker_configs import AimConfig
 
+AIM_HASH_EXPORT_FILENAME = "aim_run_hash"
+
+
+class CustomAimCallback(AimCallback):
+    """
+    Custom Aimstack callback is used to export run id from Aim
+    as soon as its created, which is on_init_end.
+    """
+
+    # path where we export run hash generated by Aim
+    # This is used to link back to the expriments from outside aimstack
+    hash_export_path = None
+
+    def on_init_end(self, args, state, control, **kwargs):
+        # pylint: disable=unused-argument
+        """
+        Override Aimstack callback on_init_end function.
+        First call AimCallback.setup to initialize internal structures
+        Then export run_id_hash to a file.
+        Run id is exported to,
+        AimConfig.aim_run_id_export_path if it is passed
+        training_args.output_dir/aim_run_hash.json if output_dir is present.
+        Its not exported if both paths are invalid.
```

Comment:
```suggestion
        Override Aimstack callback on_init_end function.
        First call AimCallback.setup to initialize internal structures.
        Then, export run_id_hash to a file.
        Run id is exported to,
        AimConfig.aim_run_id_export_path if it is passed, otherwise,
        training_args.output_dir/aim_run_hash.json if output_dir is present.
        Its not exported if both paths are invalid.
```

Reply:
changed